src/python/pants/engine/internals/graph.py

```diff
@@ -198,34 +200,18 @@ class Owners:
 async def find_owners(owners_request: OwnersRequest) -> Owners:
     sources_set = FrozenOrderedSet(owners_request.sources)
     dirs_set = FrozenOrderedSet(os.path.dirname(source) for source in sources_set)
-
-    # Walk up the buildroot looking for targets that would conceivably claim changed sources.
```

Comment:
Would be good to leave this comment, as the behavior is a bit non-intuitive. In particular, it means that we'll only find owners "above" a source file, which we should eventually enforce in some way... but at the least let's leave the comment.